### PR TITLE
lib/promscrape/discovery/kubernetes: support kubernetes native sidecars

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

Added Kubernetes Native Sidecar support
fixes https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7287

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: lib/promscrape/discovery/kubernetes/endpoints.go [hunk 1]
```diff
@@ -194,6 +204,19 @@ func getEndpointLabelsForAddressAndPort(gw *groupWatcher, podPortsSeen map[*Pod]
 			}
 		}
 	}
+	for _, c := range p.Spec.InitContainers {
+		// Defines native sidecar https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/#what-are-sidecar-containers-in-1-28
+		if c.RestartPolicy != "Always" {
```
Comment (f41gh7) [inline lib/promscrape/discovery/kubernetes/endpoints.go:209]: Btw, prometheus adds unconditionally `initContainers` to the `Containers` and I think it's a bad behavior. Your solution is better to me.

File: lib/promscrape/discovery/kubernetes/endpoints.go [hunk 2]
```diff
@@ -123,29 +123,39 @@ func (eps *Endpoints) getTargetLabels(gw *groupWatcher) []*promutils.Labels {
...
-				p.appendEndpointLabels(m, eps)
-				if svc != nil {
-					svc.appendCommonLabels(m)
-				}
-				// Remove possible duplicate labels, which can appear after appendCommonLabels() call
-				m.RemoveDuplicates()
-				ms = append(ms, m)
+			appendPodMetadata(p, &c, ports)
+		}
+		for _, c := range p.Spec.InitContainers {
```
[the remark this answers is not in the document]
Comment (f41gh7) [reply lib/promscrape/discovery/kubernetes/endpoints.go:153]: It's also required to add the same logic to the endpointslice.go

File: docs/changelog/CHANGELOG.md [hunk 3]
```diff
@@ -34,6 +34,7 @@ Released at 2024-10-21
 * FEATURE: [vmgateway](https://docs.victoriametrics.com/vmgateway/): allow parsing `scope` claim parsing in array format. This is useful for cases when identity provider does encode claims in array format.
 * FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): add the ability to cancel running queries. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7097).
 * FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): add playgrounds for [retention filters](https://docs.victoriametrics.com/single-server-victoriametrics/#retention-filters) and [downsampling](https://docs.victoriametrics.com/single-server-victoriametrics/#downsampling) configurations. These new pages are available in vmui in the `Tools` section. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6304).
+* FEATURE: [vmagent](https://docs.victoriametrics.com/vmagent/): support scraping from Kubernetes Native Sidecars. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7287).
```
Comment (zekker6) [inline docs/changelog/CHANGELOG.md:37]: FYI: seems like changelog entry needs to be moved to `tip` section

**Review by zekker6 — APPROVED** [review]
LGTM, see 1 minor comment

File: lib/promscrape/discovery/kubernetes/endpoints_test.go [hunk 4]
```diff
@@ -115,8 +115,9 @@ func TestParseEndpointsListSuccess(t *testing.T) {
 
 func TestGetEndpointsLabels(t *testing.T) {
 	type testArgs struct {
-		containerPorts map[string][]ContainerPort
-		endpointPorts  []EndpointPort
+		containerPorts     map[string][]ContainerPort
+		initContainerPorts map[string][]ContainerPort
```
Comment (f41gh7) [inline lib/promscrape/discovery/kubernetes/endpoints_test.go:119]: I suggest to introduce additional in-place type initContainerPorts:

```
type initContainerWithPorts{
  name string
  image string
  restartPolicy string
  ports []ContainerPort
}
```
It should improve test quality.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**